src/cell_actor.c

```diff
@@ -0,0 +1,935 @@
+    NNS_G2dBeginRendering(collection->renderer);
+    NNS_G2dPushMtx();
+
+    NNS_G2dSetRndCoreAffineOverwriteMode(&collection->renderer->rendererCore, actor->affineOverwriteMode);
+
+    if (actor->affineOverwriteMode == NNS_G2D_RND_AFFINE_OVERWRITE_NONE) {
+        NNS_G2dSetRndCoreFlipMode(
+            &collection->renderer->rendererCore, 
+            actor->flip & CELL_ACTOR_FLIP_H, 
+            actor->flip & CELL_ACTOR_FLIP_V
+        );
+    } else {
+        NNS_G2dSetRndCoreFlipMode(&collection->renderer->rendererCore, FALSE, FALSE);
+    }
+
+    NNS_G2dTranslate(pos.x, pos.y, pos.z);
+
+    if (actor->affineOverwriteMode != NNS_G2D_RND_AFFINE_OVERWRITE_NONE) {
+        NNS_G2dTranslate(actor->affineTranslation.x, actor->affineTranslation.y, actor->affineTranslation.z);
+        NNS_G2dScale(actor->affineScale.x, actor->affineScale.y, actor->affineScale.z);
+        NNS_G2dRotZ(FX_SinIdx(actor->affineZRotation), FX_CosIdx(actor->affineZRotation));
+
+        // affineTranslation only serves as a pivot point for rotation and scaling
+        // so we undo this translation after applying these transformations.
+        NNS_G2dTranslate(-actor->affineTranslation.x, -actor->affineTranslation.y, -actor->affineTranslation.z);
+    }
+
+    // Set the overwrite parameters
+    // We always want the actual overwrite flags to be equal to overwriteParam
+    // so we set the flags that are not in overwriteParam to 0 with the second call
+    NNS_G2dSetRendererOverwriteEnable(collection->renderer, actor->overwriteFlags);
+    NNS_G2dSetRendererOverwriteDisable(collection->renderer, ~actor->overwriteFlags);
+
+    NNS_G2dSetRendererOverwritePlttNo(collection->renderer, actor->explicitPalette);
+    NNS_G2dSetRendererOverwritePlttNoOffset(collection->renderer, actor->explicitPaletteOffset);
+    NNS_G2dSetRendererOverwriteMosaicFlag(collection->renderer, actor->explicitMosaic);
+    NNS_G2dSetRendererOverwriteOBJMode(collection->renderer, actor->explicitOamMode);
+    NNS_G2dSetRendererOverwritePriority(collection->renderer, actor->explicitPriority);
+
+    if (actor->type == CELL_ANIM_TYPE_CELL || actor->type == CELL_ANIM_TYPE_VRAM_CELL) {
+        CellAnimationData *cellAnim = (CellAnimationData *)&actor->animData;
+        NNS_G2dDrawCellAnimation(&cellAnim->anim);
+    } else {
+        MultiCellAnimationData *multiCellAnim = (MultiCellAnimationData *)&actor->animData;
+        NNS_G2dDrawMultiCellAnimation(&multiCellAnim->anim);
+    }
+
+    NNS_G2dPopMtx();
+    NNS_G2dEndRendering();
+}
+
+static void CellActorCollection_DrawActor_Stub(const CellActorCollection *collection, CellActor *actor)
+{
+    return;
+}
+
+static void CellActor_UpdateAnimInternal(CellActor *actor)
+{
+    CellActor_UpdateAnim(actor, actor->animSpeed);
+}
+
+static void CellActor_UpdateAnimInternal_Stub(CellActor *actor)
+{
+    return;
+}
+
+static void CellActorCollection_Insert(CellActorCollection *collection, CellActor *actor)
```

Comment:
**suggestion:** This code leads me to believe that this collection is effectively a priority queue. Does that make sense with helping to name it?

Reply:
Well yeah, it's pretty similar to the task system in that regard. I.. still don't know what to call it though. Personally at least I think Actor isn't bad, it's a pretty common term in gamedev. Maybe it needs something to indicate that it's a "visual thing". Idk. I'm open to changing it

Reply:
Looking at this again, I think there is just so much going on in these structs that makes them messy and vague that any term we come up with is going to be purposefully and similarly vague. In that sense, I'm fine with `actor`.